tests/test_responses_between_dates.py

```diff
@@ -0,0 +1,97 @@
+            ({"start_date": str(date.today()),"end_date": str(date.today())},[]),
+
+            #Date range covers first week of test data
+            ({"start_date": "1969-12-29","end_date": "1970-01-02"},[
+            {"respondent_id": 1, "date": "1970-01-02", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"}
+            ]),
+
+            #Date range covers first two weeks of test data
+            ({"start_date": "1969-12-29","end_date": "1970-01-09"},[
+            {"respondent_id": 1, "date": "1970-01-02", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"},
+            {"respondent_id": 1, "date": "1970-01-09", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-09", "question_no": 1, "answer": "Generous Giver"}
+            ]),
+
+            #Date range from a week prior to the dates where test data is present ,to the week succeeding the last date which has the test date
+            ({"start_date": "1969-12-23","end_date": "1970-01-23"},[
+            {"respondent_id": 1, "date": "1970-01-02", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 1, "date": "1970-01-09", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 1, "date": "1970-01-16", "question_no": 2, "answer": "Stingy Taker"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-09", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-16", "question_no": 1, "answer": "Generous Giver"},
+            {"respondent_id": 2, "date": "1970-01-02", "question_no": 2, "answer": "Generous Giver"},
+            ])
+            ]
+
+
+
+@pytest.mark.parametrize("date_range,expected_response", TEST_DATA)
+def test_data_between_dateranges(date_range,expected_response):
+    "Test the response data recieved for different date ranges"
+    
+    #Get the response from API
+    data_date_range = date_range
+    json_data = json.dumps(data_date_range)
+    response = requests.post(responses_url, data = json_data, headers = {'User': API_KEY})
+    response_data = response.json()
+    #Compare the two response recieved with the actual response
+    diff = [i for i in response_data + expected_response if i not in response_data or i not in expected_response]
+    result = len(diff) == 0
+
+    assert response.status_code == 200
+    assert len(response_data) == len(expected_response)
+    assert result == True,f'There are {len(diff)} differences:\n{diff[:5]}'
```

Comment:
Why is there a slice of diff here? The difference should just be {diff} and not {diff[:5]}

Reply:
Yes you are right, it only prints the first 5 differences, have changed it